### ISSUE TITLE
[Task][HF] 3.1: Accelerate Accelerator.prepare() compatibility

### ISSUE THREAD

**Issue by lvyufeng** [candle-org/candle#231]
## Parent Issue

Part of #204 — HF Ecosystem Roadmap, **Phase 3 (v0.3.0): Task 3.1**

## Description

Test and ensure candle works with accelerate's `Accelerator` class: `prepare(model, optimizer, dataloader)`, gradient accumulation, mixed-precision, `unwrap_model`, `save_state`/`load_state`.

## Requirements

- [ ] `Accelerator.prepare(model)` wraps candle model correctly
- [ ] `Accelerator.prepare(optimizer)` wraps candle optimizer correctly
- [ ] `Accelerator.prepare(dataloader)` wraps candle dataloader correctly
- [ ] Gradient accumulation context manager works
- [ ] Mixed-precision training via accelerate works
- [ ] `Accelerator.unwrap_model()` returns original candle model
- [ ] `Accelerator.save_state()` / `Accelerator.load_state()` work
- [ ] `Accelerator.backward(loss)` works correctly
- [ ] Device placement via accelerate works

## Success Criteria

```python
import candle as torch
from accelerate import Accelerator

accelerator = Accelerator()
model = torch.nn.Linear(10, 5)
optimizer = torch.optim.Adam(model.parameters())
model, optimizer = accelerator.prepare(model, optimizer)
# Training loop works
```

## Blocked By

- Phase 1 and Phase 2 tasks

## Blocks

- Required for HF Trainer and TRL integration

**Comment by holysheep123** [comment]
Hi **lvyufeng** 鈥?happy to help debug. To speed things up, could you share:

1. The exact error message (full text, not just the code)
2. What model/endpoint you're trying to use
3. Whether this was working before or is a fresh setup

鈥?holysheep.ai
